src/server/snapshot.cc

```diff
@@ -185,7 +185,7 @@ void SliceSnapshot::SerializeSingleEntry(DbIndex db_indx, const PrimeKey& pk, co
 
 bool SliceSnapshot::FlushSfile(bool force) {
   if (force) {
-    auto ec = rdb_serializer_->FlushMem();
+    auto ec = rdb_serializer_->FlushToSink(sfile_.get());
```

Comment:
I think `sfile_->val.size()` condition below will always be below 4096 so `FlushSfile(false)` will never flush and just aggregate the whole snapshot internally.

Reply:
you are so right